### PR TITLE
71096 sole trader

### PR DESCRIPTION
**Pull request by stevelee-civica**
71096 add sole trader screen
add unit tests
update navigations

### CODE REVIEW

File: src/EA.Weee.Core/Organisations/SoleTraderViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿namespace EA.Weee.Core.Organisations
+{
+    using System.ComponentModel;
+    using System.ComponentModel.DataAnnotations;
+    using EA.Weee.Core.DataStandards;
+    public class SoleTraderViewModel
+    {
+        [Required]
+        [DisplayName("First name")]
+        [StringLength(CommonMaxFieldLengths.DefaultString)]
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Core/Organisations/SoleTraderViewModel.cs:10]: @stevelee-civica have a look at the latest partner view model as I think they use different length fields, think there is a Firstname and lastname one

**Comment by JohnPadgettCivica** [comment]
@stevelee-civica the organisation transaction service will need to be updated to persist this data I believe. 

Also as we now have this new screen in between the Type screen and Organisation details screen, clicking the Back link from organisation details will need to take us to the Partner details screen.

This line 
image
Will have to be conditional based on the Model.OrganisationType value. It also looks broken for Partner.

File: src/EA.Weee.Web.Tests.Unit/Controllers/OrganisationRegistrationControllerTests.cs [hunk 2]
```diff
@@ -1435,5 +1435,42 @@ public async Task PartnerDetails_Post_AddErrorIfMoreThanTenPartners()
 
             var model = result.Model as PartnerViewModel;
         }
+
+        [Fact]
+        public async Task SoleTraderDetails_Get_ReturnsView()
+        {
+            // Act
+            var result = await controller.SoleTraderDetails() as ViewResult;
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web.Tests.Unit/Controllers/OrganisationRegistrationControllerTests.cs:1443]: @stevelee-civica There should also be a test to check that when the 

var existingTransaction = await transactionService.GetOrganisationTransactionData(User.GetAccessToken());

Call is made in the GET method that that existing model is returned to the view.

Comment (stevelee-civica) [reply src/EA.Weee.Web.Tests.Unit/Controllers/OrganisationRegistrationControllerTests.cs:1443]: added unit test

File: src/EA.Weee.Core/Organisations/SoleTraderViewModel.cs (hunk 1 above)
Comment (stevelee-civica) [reply src/EA.Weee.Core/Organisations/SoleTraderViewModel.cs:10]: updated

File: src/EA.Weee.Web.Tests.Unit/Controllers/OrganisationRegistrationControllerTests.cs (hunk 2 above)
Comment (stevelee-civica) [reply src/EA.Weee.Web.Tests.Unit/Controllers/OrganisationRegistrationControllerTests.cs:1533]: updated organisation transaction service to handle SoleTraderViewModel

**Pull request merged**